### ISSUE TITLE
look at upgrading go to 1.12.9 / 1.11.13

### ISSUE THREAD

**Issue by BenTheElder** [kubernetes/kubernetes#79912]
These are out now, announcement here:
https://groups.google.com/forum/#!msg/golang-announce/-JidGVRIVEc/MAGRIyQoCgAJ

@kubernetes/release-team 
@kubernetes/release-engineering

**Comment by BenTheElder** [comment]
For 1.12.7:

> These releases include fixes to cgo, the compiler, and the linker. 

https://github.com/golang/go/issues?q=milestone%3AGo1.12.7

----

/sig release
(also maybe scale??)

**Comment by BenTheElder** [comment]
looks like pretty much the same fixes to 1.11.12

**Comment by justaugustus** [comment]
Thanks for raising this, @BenTheElder. We'll look into it for the 1.16 cycle.
/milestone v1.16
/area release-team release-eng
/priority important-soon

**Comment by tao12345666333** [comment]
I have submitted a PR for this. #79966

**Comment by tao12345666333** [comment]
#79966 has been merged. Maybe we can close this.

**Comment by justaugustus** [comment]
@tao12345666333 -- we also need to update the other supported release branches: `release-1.15`, `release-1.14`, and `release-1.13`. Would you like to work on that as well?

**Comment by tao12345666333** [comment]
yes. I will work on that.

**Comment by simplytunde** [comment]
y
How
On Fri, Jul 12, 2019 at 9:50 I PM Jintao Zhang <notifications@github.Icom>
How wrote:

**Comment by tao12345666333** [comment]
PR for release-1.15 #80134

**Comment by neolit123** [comment]
i don't see anything critical in the changelog for this release, why are we bumping golang in the older k8s branches?

**Comment by cblecker** [comment]
@neolit123 see #73286

There are lots of reasons, but the biggest one in my mind is we need to keep them current because of the possibility of a future security fix. We want to start performance/regression testing as soon as we can, because if there is a problem we need to report to the go team, we want to do that before we're forced to accept a performance regression because of a critical security fix.

**Comment by dims** [comment]
This is the right time in the cycle to do this for another reason :)

**Comment by tpepper** [comment]
Bump...

Do we really consider this a kind feature?  I'd see it a bug fixing to stay up to date with critical tool chain bug fixes.

Need a determination on the 1.15 branch move: https://github.com/kubernetes/kubernetes/pull/80134

**Comment by cblecker** [comment]
It's tricky because it's somewhere between the two IMO. I don't care so much about the labels, but I'm clear that we need to move forward. If anyone has any strong objections, please raise them now!

- `master` (1.16) is on 1.12.7
- I've just approved the PR for `release-1.15` to move to 1.12.7
- `release-1.14` needs to move to 1.12.7
- `release-1.13` needs to move to 1.11.12

If anyone wants to pick up an additional branch, please comment here and let me know. The PRs for `master` and `release-1.15` can be used as templates.

/good-first-issue

**Comment by warmchang** [comment]
> It's tricky because it's somewhere between the two IMO. I don't care so much about the labels, but I'm clear that we need to move forward. If anyone has any strong objections, please raise them now!
> 
> * `master` (1.16) is on 1.12.7
> * I've just approved the PR for `release-1.15` to move to 1.12.7
> * `release-1.14` needs to move to 1.12.7
> * `release-1.13` needs to move to 1.11.12
> 
> If anyone wants to pick up an additional branch, please comment here and let me know. The PRs for `master` and `release-1.15` can be used as templates.
> 
> /good-first-issue

@cblecker Please pay attention to this issue (https://github.com/golang/go/issues/33405 cause fd escape to child process) with go1.12.7.

It's fixed by go team on [release-branch.go1.12](https://go-review.googlesource.com/c/go/+/188538/), but still need to wait the v1.12.8 release.

**Comment by cblecker** [comment]
@warmchang Thanks for the heads up. I don't want that to block our move. We should be moving on these a little more rapidly. Until 1.12.8 is out, we should update things to 1.12.7, and then can push forward to 1.12.8 when it is released.

**Comment by buribae** [comment]
@cblecker Can I work on release-1.14?

**Comment by cblecker** [comment]
@buribae Go for it.
/remove-help

**Comment by BenTheElder** [comment]
1.12.8 is out https://groups.google.com/forum/#!topic/golang-announce/65QixT3tcmg

**Comment by cblecker** [comment]
update master to go 1.12.8: https://github.com/kubernetes/kubernetes/pull/81390

**Comment by buribae** [comment]
update release-1.14 to go 1.12.8: #81392

**Comment by cblecker** [comment]
Once the master PR merges, we will need to do the following:
- Cherry pick #81390 back to release-1.15 (@imkin will handle this)
- Cherry pick #81390 back to release-1.14 (@buribae can handle this if they would like)
- A fresh PR opened against release-1.13, bumping it to go 1.11.13
- test-infra golang versions also need to be updated

**Comment by warmchang** [comment]
> Once the master PR merges, we will need to do the following:
> 
> * Cherry pick #81390 back to release-1.15 (@imkin will handle this)
> * Cherry pick #81390 back to release-1.14 (@buribae can handle this if they would like)
> * A fresh PR opened against release-1.13, bumping it to go 1.11.13
> * test-infra golang versions also need to be updated

I'll deal with the release-1.13, bumping it to go 1.11.13

**Comment by buribae** [comment]
> Once the master PR merges, we will need to do the following:
> 
> * Cherry pick #81390 back to release-1.15 (@imkin will handle this)
> * Cherry pick #81390 back to release-1.14 (@buribae can handle this if they would like)
> * A fresh PR opened against release-1.13, bumping it to go 1.11.13
> * test-infra golang versions also need to be updated



> Once the master PR merges, we will need to do the following:
> 
> * Cherry pick #81390 back to release-1.15 (@imkin will handle this)
> * Cherry pick #81390 back to release-1.14 (@buribae can handle this if they would like)
> * A fresh PR opened against release-1.13, bumping it to go 1.11.13
> * test-infra golang versions also need to be updated

sure. I'll cherry-pick.

**Comment by tao12345666333** [comment]
> @warmchang Thanks for the heads up. I don't want that to block our move. We should be moving on these a little more rapidly. Until 1.12.8 is out, we should update things to 1.12.7, and then can push forward to 1.12.8 when it is released.

@cblecker I have opened a PR #81402 to bump to 1.12.7. Should we need do it first?

**Comment by cblecker** [comment]
@tao12345666333 No, we don't need to do it as a separate PR.

When @buribae opens the cherry-pick after #81390 merges, they can pick both PRs back at the same time with the following command:
```
hack/cherry_pick_pull.sh upstream/release-1.14 79966 81390
```

**Comment by warmchang** [comment]
Hi @cblecker 

There is no kube-cross:v1.11.13-1 image in the gcr regitstry, and I'v not the right to make & push it.

https://console.cloud.google.com/gcr/images/google-containers/GLOBAL/kube-cross?gcrImageListsize=30

Do you know who can make this? Thanks.

**Comment by tao12345666333** [comment]
> There is no kube-cross:v1.11.13-1 image in the gcr regitstry, and I'v not the right to make & push it.
> 
> https://console.cloud.google.com/gcr/images/google-containers/GLOBAL/kube-cross?gcrImageListsize=30
> 
> Do you know who can make this? Thanks.

@javier-b-perez PTAL. Thanks. :smile_cat:

**Comment by cblecker** [comment]
The bazel issues on #81398 need to be fixed first.

**Comment by cblecker** [comment]
@imkin You are clear to pick #81394 back to release-1.15
@buribae You are clear to pick #81394 back to release-1.14 (see the command I posted in https://github.com/kubernetes/kubernetes/issues/79912#issuecomment-521096598)

**Comment by buribae** [comment]
@cblecker I do not think go.sum and go.mod existed in release-1.14. What should we do?

**Comment by cblecker** [comment]
@buribae Sorry, I meant #81390, not #81394

**Comment by buribae** [comment]
PR for Cherry pick #81390 back to release-1.14 #81449

**Comment by imkin** [comment]
@cblecker cherry-pick is here. 
https://github.com/kubernetes/kubernetes/pull/81487

**Comment by imkin** [comment]
@cblecker are we supposed to have the build_dependencies.yaml in 1.15 release?

**Comment by cblecker** [comment]
go1.12.9 just got released today. We may have revisit this plan.

**Comment by BenTheElder** [comment]
https://github.com/kubernetes/kubernetes/pull/81489 brought Kubernetes master to go 1.12.9
https://github.com/kubernetes/kubernetes/pull/81542 for Kubernetes 1.13 to go 1.11.13
/assign
/assign @cblecker

**Comment by BenTheElder** [comment]
/priority critical-urgent
bumping given network CVEs in go

**Comment by warmchang** [comment]
> #81489 brought Kubernetes master to go 1.12.9
> #81542 for Kubernetes 1.13 to go 1.11.13
> /assign
> /assign @cblecker

Duplicate PR?
https://github.com/kubernetes/kubernetes/pull/81398

**Comment by BenTheElder** [comment]
> Duplicate PR?
> #81398

#81398 is incomplete.

**Comment by BenTheElder** [comment]
https://github.com/kubernetes/test-infra/pull/13957 for the test-infra images, though it will need a follow-up or two to actually deploy the new images (kubekins-e2e should get auto-bumped with prow, kubekins-test requires manual updates but isn't relevant for kubernetes >= 1.15)

**Comment by cblecker** [comment]
Status:
- master, release-1.15, and release-1.14 have all had both their golang, and the golang.org/x/net dependency updated
- test-infra images for master, release-1.15, release-1.14, and release-1.13 have all been updated
- #81542 is pending to update the golang and bazel versions on release-1.13
- #81546 is pending to update the golang.org/x/net dependency on release-1.13

**Comment by BenTheElder** [comment]
> test-infra images for master, release-1.15, release-1.14, and release-1.13 have all been updated

[Clarifying that the image definitions were updated, but the general / e2e images have not rolled out yet. The verify / integration images for Kubernetes ≤ 1.14 have though]

**Comment by justaugustus** [comment]
(golang changes have also been pulled into `release-1.16` with [the latest branch fast-forward](https://github.com/kubernetes/kubernetes/compare/a5110a8849...f142fb7df5).)

**Comment by cblecker** [comment]
We completed this upgrade over the weekend. Thank you to all involved.

To those that attempted PRs, but were unable to be merged: We had to start moving faster because of the security context of 1.12.8, but I'm glad there are others who are interested in this part of the project. I hope you will all attempt again the next time we bump these versions. Feel free to reach out to me in the future if you need assistance/mentoring in doing this.
